### PR TITLE
Edit Point Radius MGRS coordinates

### PR DESCRIPTION
**Pull request by bdeining**
Changing the validation logic to no longer clear the value
  Changing the validation logic to default lat/lon to 0 instead of undefined
  Adding a Warning to the user when an invalid MGRS coordinate is entered
  Changing the logic that converts from Lat/Lon to USNG to run after drawing is complete

#### What are the relevant tickets?
Fixes: #____

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by bdeining** [comment]
build now

**Comment by bdeining** [comment]
build now

File: ui/packages/catalog-ui-search/src/main/webapp/component/location-old/location-old.js [hunk 1]
```diff
@@ -76,7 +76,8 @@ function convertToValid(key, model) {
     key.lon = Math.min(180, key.lon)
   }
   if (key.radius !== undefined) {
-    key.radius = Math.max(minimumBuffer, key.radius)
+    let tempRadius = Math.max(minimumBuffer, key.radius)
+    key.radius = isNaN(tempRadius) ? model.get('radius') : tempRadius
```
Comment (willwill96) [inline ui/packages/catalog-ui-search/src/main/webapp/component/location-old/location-old.js]: :pencil2: The logic here seems strange to me, but this is equivalent:
```suggestion
    if (isNaN(key.radius)) {
        key.radius = model.get('radius')
    }
```

File: ui/packages/catalog-ui-search/src/main/webapp/component/location-old/location-old.js [hunk 2]
```diff
@@ -639,10 +640,8 @@ module.exports = Backbone.AssociatedModel.extend({
     } else {
       this.clearUtmUpsPointRadius(true)
```
Comment (willwill96) [inline ui/packages/catalog-ui-search/src/main/webapp/component/location-old/location-old.js]: Should we still be calling this function?

File: ui/packages/catalog-ui-search/src/main/webapp/component/location-old/location-old.js (hunk 1 above)
Comment (millerw8) [reply ui/packages/catalog-ui-search/src/main/webapp/component/location-old/location-old.js]: hm, not exactly... that says if radius is a number don't enforce the min value.

But the original seems a little funny too, we check if it is a number after we do the max check... according to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/max if key.radius is NaN, tempRadius will also be NaN

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/location/point-radius.js [hunk 3]
```diff
@@ -58,14 +72,30 @@ const PointRadiusLatLon = props => {
   )
 }
 
+const usngs = require('usng.js')
```
Comment (Schachte) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/location/point-radius.js]: ✏️Require statement at the top of file

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/location/point-radius.js [hunk 4]
```diff
@@ -58,14 +72,30 @@ const PointRadiusLatLon = props => {
...
   return (
     <div>
       <TextField label="USNG / MGRS" value={usng} onChange={cursor('usng')} />
       <Units value={radiusUnits} onChange={cursor('radiusUnits')}>
         <TextField label="Radius" value={radius} onChange={cursor('radius')} />
       </Units>
+      {error ? (
+        <ErrorBlock>
+          <WarningIcon className="fa fa-warning" />
+          <span>Invalid USNG / MGRS coords</span>
```
Comment (Schachte) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/location/point-radius.js]: ✏️ coords to coordinates

File: ui/packages/catalog-ui-search/src/main/webapp/component/location-old/location-old.js (hunk 1 above)
Comment (bdeining) [reply ui/packages/catalog-ui-search/src/main/webapp/component/location-old/location-old.js]: are we going to hold the pr up for this?

**Comment by bdeining** [comment]
hero'd here https://github.com/codice/ddf/pull/5497

File: ui/packages/catalog-ui-search/src/main/webapp/component/location-old/location-old.js [hunk 5]
```diff
@@ -76,7 +76,8 @@ function convertToValid(key, model) {
     key.lon = Math.min(180, key.lon)
   }
   if (key.radius !== undefined) {
-    key.radius = Math.max(minimumBuffer, key.radius)
+    let tempRadius = Math.max(minimumBuffer, key.radius)
```
Comment (jrnorth) [inline ui/packages/catalog-ui-search/src/main/webapp/component/location-old/location-old.js]: ✏️ `const`
Also, could we name this something that does more to distinguish it from `model.get('radius')`? I don't know what the difference between the two is just by looking at the code.

**Comment by bdeining** [comment]
build now

**Comment by bdeining** [comment]
build now

**Pull request merged**